lib/collection/src/collection/shard_transfer.rs

```diff
@@ -182,6 +183,7 @@ impl Collection {
     pub async fn finish_shard_transfer(
         &self,
         transfer: ShardTransfer,
+        all_peers: &HashSet<PeerId>,
         shard_holder: Option<&ShardHolder>,
     ) -> CollectionResult<()> {
```

Comment:
Here, can't we just check whether the peer ID is in the shard holder?

Because it's just when finishing a shard transfer, I think that should be fine. We don't expect to create shards on new peers here.

Reply:
Hmmmm.... this should work as well, I think.